### PR TITLE
 [speechlm2] SALM with TDT decoder input; "sharegpt" format support; various improvements

### PR DESCRIPTION
**Pull request by pzelasko**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

SALM with TDT decoder input; "sharegpt" format support; various improvements

**Collection**: speechlm2

# Changelog

* `SALMWithAsrDecoder` class, feeds LLM with acoustic embeddings + transcript from pretrained ASR model (intended for parakeet-tdt only atm)
* Two new modality adapters: multi-layer linear projection (includes intermediate encoder representations) and QFormer
* Refresh default configs for speechlm2 SALM models
* Bug fixes in `salm_eval.py` and `salm_generate.py`
* Support `share_gpt` data format type
* Reduce log verbosity when creating multi-dataset dataloader
* [internal related] support converting legacy spoken question answering data format to NeMoMultimodalConversation
* Support dynamic mono downmixing of stereo examples for SALM training
* Contribute preliminary OOMptimizer script for speechlm2 SALM

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/common/data/lhotse/cutset.py [hunk 1]
```diff
@@ -752,6 +767,50 @@ def read_lhotse_as_conversation(config) -> tuple[CutSet, bool]:
     return cuts, is_tarred
 
 
+def sqa_cut_to_conversation(
+    cut: Cut, audio_locator_tag: str, token_equivalent_duration: float
+) -> NeMoMultimodalConversation:
+    """
+    Converts a lhotse Cut representing a SQA example into a NeMoMultimodalConversation.
+
```
Comment (zhehuaichen) [inline nemo/collections/common/data/lhotse/cutset.py:804]: can we include an example jsonl to show the format here

File: scripts/speechlm2/auto_launcher_with_seed.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
```
Comment (zhehuaichen) [inline scripts/speechlm2/auto_launcher_with_seed.sh]: is it necessary to check in this?

**Review by zhehuaichen — DISMISSED** [review]

File: scripts/speechlm2/auto_launcher_with_seed.sh (hunk 2 above)
Comment (pzelasko) [reply scripts/speechlm2/auto_launcher_with_seed.sh]: technically not, will remove

**Comment by chtruong814** [comment]
The CI failed due in the last step because the Github hosted runners are having problems.  All the CI jobs on the self-hosted runners passed.  Will go ahead and merge this.
